### ISSUE TITLE
Remove ultralytics dependency and run inference file for python version 3.6

### ISSUE THREAD

**Issue by Gcc1012** [ultralytics/ultralytics#20620]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

 I have trained a classification model using yolo11n , now i have to write its inference script , but the problem is for development and deployment i have python version 3.6, but yolo and ultralytics supports python version 3.8 and above only , i have tried to convert the classification model i.e .pt file into torchscript to remove this dependency issue , but converting into torchscript is lowering the accuracy of the model , is there any way , or any souce i can refer to resolve this issue? My requirement is to use .pt file directly but also by removing the dependency of ultralytics

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Gcc1012, thank you for reaching out and for your interest in Ultralytics 🚀! This is an automated response to help you get started, and an Ultralytics engineer will assist you soon.

We recommend visiting the [Docs](https://docs.ultralytics.com/) for guidance, where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, including information about model export and inference options. Many common questions are already addressed there.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us investigate further.

If this is a custom training ❓ Question, please include as much detail as possible, such as your dataset image examples, training logs, and confirm you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community wherever you prefer! For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. For in-depth discussions, see [Discourse](https://community.ultralytics.com/). Or join the conversation on our [Subreddit](https://reddit.com/r/Ultralytics).

## Upgrade

Upgrade to the latest `ultralytics` package with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to ensure your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of these up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
You can export to ONNX with `dynamic=True`

**Comment by Gcc1012** [comment]
I dont want to convert it in any other format , i just want to use .pt file , my requirement is to load the .pt file using torch , no dependency of ultralytics

**Comment by Y-T-G** [comment]
You can't load `.pt` without the original definitions. That's how `.pt` works.

**Comment by Gcc1012** [comment]
we can load by rebuilding model using the architeture of trained model , but it is a lengthy process, i need a predefined or easy solution for it

**Comment by Y-T-G** [comment]
The easiest solution is to export to a different format. Because you can't load `.pt` files without the original module definitions. The module definitions are in Ultralytics. So you need Ultralytics if you want to use `.pt`.

**Comment by Gcc1012** [comment]
is there any way that we can accumulate all the module definitions like combine them from ultralytics and use them we rebuilded model  using the architecture of yolo11

**Comment by glenn-jocher** [comment]
While it's technically possible to extract all the required module definitions from Ultralytics and incorporate them into a standalone project, this is not recommended as it would be extremely complex and error-prone. You'd need to track all dependencies, class definitions, and implementation details across multiple files.

Your best options are:

1. Export to ONNX format with `model.export(format="onnx", dynamic=True)` - ONNX typically maintains accuracy better than TorchScript and has excellent cross-platform compatibility.

2. If you're concerned about accuracy loss, try fine-tuning the export parameters like `simplify=True` or adjusting the ONNX opset version.

3. If you absolutely must use the original architecture, consider setting up a separate Python 3.8+ environment just for inference with Ultralytics.

Export formats are specifically designed for deployment scenarios like yours where the original framework dependencies need to be removed.
